### ISSUE TITLE
project discussion, now I guess?

### ISSUE THREAD

**Issue by jtagcat** [jtagcat/samba#2]
samba fully crashes when windows client with mounted drive deletes a .lnk  
oops  
dunno if reported

the helpful link in log is very helpful https://wiki.samba.org/index.php/Bug

gonna go sleep  
seriously, fuck nextcloud  
windows is garbage as well  
so is software
and humans writing it
go to bed

**Comment by cruizba** [comment]
hahaha nice issue rant. Thanks for the fork anyway, I was thinking to fork it when I see it was not maintained nor updated.

I'm agree with you, software is a mess nowadays. Too much complexity. We will pay it in the future.

Best regards

**Comment by cruizba** [comment]
Just FYI, I end up using this fork (https://github.com/Erriez/docker-samba) because I was unable to use it on arm64. Looks pretty good too, and it looks like the author wants to maintain it. I was thinking about making a pull request with autobuilds with github actions to keep images always updated.

**Comment by jtagcat** [comment]
> I was unable to use it on arm64
Saying that isn't enough, do give details, did you get an error message or something?

Checked it out myself:
```
$ docker run --rm ghcr.io/jtagcat/samba:armhf
Unable to find image 'ghcr.io/jtagcat/samba:armhf' locally
armhf: Pulling from jtagcat/samba
59bf1c3509f3: Pull complete
b6b927d588d9: Pull complete
63a518eb1790: Pull complete
Digest: sha256:564bfcb16aa4869101946778b45e93909f848e80c5b9dc23aec7c708de8f2ea7
Status: Downloaded newer image for ghcr.io/jtagcat/samba:armhf
WARNING: The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested
standard_init_linux.go:228: exec user process caused: exec format error
$ docker run --rm ghcr.io/jtagcat/samba:aarch64
Unable to find image 'ghcr.io/jtagcat/samba:aarch64' locally
aarch64: Pulling from jtagcat/samba
Digest: sha256:564bfcb16aa4869101946778b45e93909f848e80c5b9dc23aec7c708de8f2ea7
Status: Downloaded newer image for ghcr.io/jtagcat/samba:aarch64
WARNING: The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested
standard_init_linux.go:228: exec user process caused: exec format error
```

Seems you are expected to use buildx, I can't quickly easily test the CI without breaking prod. I might try later, though.

It makes sense (painfully obviously so), you can't really 'just' build it on the same arch. It'd be nice if GH would make multiarch runners available for native builds.

I deleted the tags for now, as to not advertise working brokenness.

***

@Erriez are you interested in (fully declaring a fork and) maintaining the thing? I can go ahead and merge / PR relevant changes to your fork.

https://github.com/Erriez/docker-samba/pull/1 seems to be sane changes, but how do you organize builds now? Do you manually run commands (could they be documented)? Is there external CI that does that?

#TODO: link [smb.conf man](https://www.samba.org/samba/docs/current/man-html/smb.conf.5.html) somewhere, for ease of access.

***

All I can say is that I don't want to take responsibility on maintaining it. I just want to mount a god damn directory under Windows, and if possible, keep a local cache.  
Still, recycle bin just doesn't work[^1], and once in a while, the container decides to halt. One cause was the .lnk files, otherwise, I haven't had the chance to debug it enough. xattrs.. also seem broken.
I can and prefer to fix stuff for others as well.

[^1]: Tried different options with smb, `smb_set_file_dosmode: file_set_dosmode of <library folder, such as Desktop>/$RECYCLE.BIN failed: Function not implemented` still pops up.

**Comment by Erriez** [comment]
> Just FYI, I end up using this fork (https://github.com/Erriez/docker-samba) because I was unable to use it on arm64. Looks pretty good too, and it looks like the author wants to maintain it. I was thinking about making a pull request with autobuilds with github actions to keep images always updated.

@cruizba Thanks for your feedback. Yes, Github Actions autobuild is on my Todo list.

@jtagcat My plan is to start the Github Actions autobuild on every PR and automatically push multi-arch images to DockerHub when merging a PR into master.

FYI: I created a [buildx test project](https://github.com/Erriez/docker-buildx-multiarch-example) and need some time to integrate into Samba fork. When Github Actions CI is up and running, I can accept PR's. How does this sound?

**Comment by jtagcat** [comment]
For a boilerplate, you can take my .github/workflows/build-ghcr.yml file, it pushes to ghcr. I say it's much easier to deal with (auth + rate limits / auth walls + links).

In my experience, doing logic inside the CI file gets complex fast. Adding logic for PRs is dumber, if you duplicate the file, and push to a tag say pr_username_branch (names might need to be transformed). It does duplicate some content (what you may choose to store inside, say shell files). Retagging on merge to master is likely not worth your time.

I guess tag me when you want PRs, or you can probably just merge them yourself as well.

**Comment by Erriez** [comment]
My Github space reached limits, so I need a different platform for image distribution such as Dockerhub. I'm working on a different project as well which will use buildx and push to Dockerhub.

**Comment by jtagcat** [comment]
jfyi
> GitHub Packages is free for public repositories

Pretty much all GitHub and GitLab features are free under the condition that the repo is public.

**Comment by Erriez** [comment]
Github storage is limted https://github.com/pricing and I have a lot of private repositories.

**Comment by jtagcat** [comment]
Public and private repositories are not connected. You can be at your quota on private, while still having unlimited on public repos. :)

**Comment by Erriez** [comment]
Good catch! I was not aware of this. I've buildx with Github Actions up and running now in my [example project](https://github.com/Erriez/docker-buildx-multiarch-example) and I'm backporting to [docker-samba](https://github.com/Erriez/docker-samba).

**Comment by Erriez** [comment]
FYI: I've released version 1.0.0 and I'm happy to accept PR's.

**Comment by jtagcat** [comment]
@cruizba are you or anyone else interested in an updated (still i-do-not-plan-on-maintaining) version?  
@Erriez's fork poofed, probably due to [licencing infringements](https://web.archive.org/web/20220101221232/https://github.com/Erriez/docker-samba/pull/6) against @dperson and all other contributors, likely by themselves.

I'm not interested in samba myself anymore, but I do have all the commits, it can be reworked.

**Comment by Erriez** [comment]
Stop complaining @jtagcat. I've deleted my fork and will not continue to support a dead project.

**Comment by cruizba** [comment]
Umm I have not enough time to maintain such project. I think that your github workflow and weekly updates just fit my needs at the moment. Also don't feel like this is an obligation, it is ok to not maintaint, you just forked the repo, nothing else ;)

**Comment by cruizba** [comment]
@jtagcat I've made you a pull request if you want to have your own autobuild multiarch CI github action: https://github.com/jtagcat/samba/pull/3
